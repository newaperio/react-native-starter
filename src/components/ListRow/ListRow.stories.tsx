import React from "react"
import { storiesOf } from "@storybook/react-native"
import { Alert, FlatList, Text } from "react-native"

import ListRow from "."

const cryptids = [
  "Barmanou",
  "Chuchunya",
  "Chupacabra",
  "Dobhar-chú",
  "Lizard Man of Scape Ore Swamp",
  "Mamlambo",
  "Mothman",
  "Ningen",
  "Thunderbird",
]

storiesOf("ListRow", module)
  .add("default", () => {
    const renderItem = ({ item }: { item: string }) => (
      <ListRow text={item} onPress={() => Alert.alert(item)} key={item} />
    )
    return <FlatList data={cryptids} renderItem={renderItem} />
  })
  .add("with custom icon", () => {
    const renderItem = ({ item }: { item: string }) => (
      <ListRow
        text={item}
        onPress={() => Alert.alert(item)}
        key={item}
        icon={<Text style={{ opacity: 0.4 }}>Go →</Text>}
      />
    )
    return <FlatList data={cryptids} renderItem={renderItem} />
  })
