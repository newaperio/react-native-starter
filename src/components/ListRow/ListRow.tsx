import { FontAwesomeIcon } from "@fortawesome/react-native-fontawesome"
import React from "react"
import { faChevronRight } from "@fortawesome/free-solid-svg-icons"
import { Text, TouchableOpacity, View } from "react-native"

import { tw } from "~theme"

interface Props {
  onPress: () => void
  text: string
  icon?: React.ReactNode
}

export const ListRow: React.FC<Props> = props => {
  return (
    <TouchableOpacity onPress={props.onPress}>
      <View
        style={tw`bg-white border-b border-gray flex-row items-center justify-between px-3 py-4`}>
        <Text style={tw`font-body`}>{props.text}</Text>
        {props.icon || (
          <FontAwesomeIcon icon={faChevronRight} style={tw`text-stone`} />
        )}
      </View>
    </TouchableOpacity>
  )
}
