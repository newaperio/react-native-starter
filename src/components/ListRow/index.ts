export { ListRow as default } from "./ListRow"
