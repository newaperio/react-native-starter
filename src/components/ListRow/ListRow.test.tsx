import React from "react"
import { Text } from "react-native"
import renderer from "react-test-renderer"

import ListRow from "."

const onPress = () => null

test("renders correctly", () => {
  const tree = renderer
    .create(<ListRow text="Snapshot" onPress={onPress} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})

test("renders custom icon", () => {
  const tree = renderer
    .create(
      <ListRow text="Snapshot" onPress={onPress} icon={<Text>Go â†’</Text>} />,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
