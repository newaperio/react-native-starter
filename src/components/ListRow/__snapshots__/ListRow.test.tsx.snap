// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#f5f5f5",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "DM Sans",
        }
      }
    >
      Snapshot
    </Text>
    <RNSVGSvgView
      align="xMidYMid"
      bbHeight={16}
      bbWidth={16}
      fill="#999"
      focusable={false}
      height={16}
      meetOrSlice={0}
      minX={0}
      minY={0}
      secondaryFill={null}
      secondaryOpacity={0.4}
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
            "borderWidth": 0,
          },
          Object {},
          Object {
            "flex": 0,
            "height": 16,
            "width": 16,
          },
        ]
      }
      vbHeight={512}
      vbWidth={320}
      width={16}
    >
      <RNSVGGroup
        fill={4288256409}
        propList={
          Array [
            "fill",
          ]
        }
      >
        <RNSVGPath
          d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
          fill={4288256409}
          fillOpacity={1}
          propList={
            Array [
              "fill",
              "fillOpacity",
            ]
          }
        />
      </RNSVGGroup>
    </RNSVGSvgView>
  </View>
</View>
`;

exports[`renders custom icon 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#fff",
        "borderBottomWidth": 1,
        "borderColor": "#f5f5f5",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingBottom": 16,
        "paddingLeft": 12,
        "paddingRight": 12,
        "paddingTop": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "DM Sans",
        }
      }
    >
      Snapshot
    </Text>
    <Text>
      Go â†’
    </Text>
  </View>
</View>
`;
